client/my-sites/upgrades/checkout-thank-you/index.jsx

```diff
@@ -163,7 +163,7 @@ var CheckoutThankYou = React.createClass( {
 	 * @returns {*[]} an array of varying size with the component instance, then an optional purchase object possibly followed by a domain name
 	 */
 	getComponentAndPrimaryPurchaseAndDomain: function() {
-		if ( this.isDataLoaded() ) {
+		if ( this.isDataLoaded() && this.props.receiptId ) {
```

Comment:
> After all, if there's no receiptId, then it can't ever be loaded...

That's not true. We need to be able to display a generic `Thank You` page in the case no receipt was returned by the API.